common/plugins/org.jboss.tools.common.core/src/org/jboss/tools/common/core/jandex/JandexUtil.java

```diff
@@ -0,0 +1,71 @@
+/******************************************************************************* 
+ * Copyright (c) 2015 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/
+package org.jboss.tools.common.core.jandex;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Enumeration;
+import java.util.StringTokenizer;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.Index;
+import org.jboss.jandex.Indexer;
+import org.jboss.tools.common.core.CommonCorePlugin;
+
+public class JandexUtil {
+
+	public static Index createJarIndex(File jarFile, Indexer indexer) throws IOException {
```

Comment:
Can we hide what we do not wish to expose in this way:
public static Object createJarIndex(File jarFile, Object indexer)
The method will check indexer and if it is org.jboss.jandex.Indexer, then the method will return an instance of org.jboss.jandex.Index, otherwise it will return null or better throw IllegalArgumentException?
Jandex API is just a few methods, I would not like to wrap it in 'our' own API that we will have to change any time we need more of Jandex.